newrelic-agent/src/main/java/com/newrelic/agent/config/AgentConfigImpl.java

```diff
@@ -599,29 +599,19 @@ public <T> T getValue(String path) {
     }
 
     @Override
-    @SuppressWarnings("unchecked")
     public <T> T getValue(String path, T defaultValue) {
         Object value = flattenedProperties.get(Strings.replaceDotHyphenWithUnderscore(path));
         if (value == null) {
             value = flattenedProperties.get(path);
         }
-
         if (value == null) {
             return defaultValue;
         } else if (value instanceof ServerProp) {
             value = ((ServerProp) value).getValue();
             return castValue(path, value, defaultValue);
-        } else if (value instanceof String && defaultValue instanceof Boolean) {
```

Comment:
This coercion to Boolean or Integer has been shifted out to the ParsedConfigValue class